fabric-object-builders-v0/src/main/java/net/fabricmc/fabric/impl/object/builder/TradeOfferFactories.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.object.builder;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.village.TradeOffer;
+import net.minecraft.village.TradeOffers;
+
+public final class TradeOfferFactories {
```

Comment:
I suggest merging 3 subinterfaces into 1 interface and convert this into an interface (like how ClientSideNetworkRegistry has an `INSTANCE` field that is populated by a ClientSideNetworkRegistryImpl) than having 3 impl instances in 3 public static final fields in the 3 subinterfaces. Also makes it easier for users.